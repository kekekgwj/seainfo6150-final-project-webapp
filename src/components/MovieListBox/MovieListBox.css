.movie-list-box-container {
    height: 426px;
    width: 100%;
}

.snippet-list-wrapper {
    display: flex;
    flex-wrap: wrap;
}

.movie-list-box-title {
    font-size: 18px;
    color: #111;
    padding-bottom: 10px;
    border-bottom: 1px solid #eaeaea;
    margin-bottom: 18px;
}