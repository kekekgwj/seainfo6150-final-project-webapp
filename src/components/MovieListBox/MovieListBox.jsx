import React from "react";
import "./MovieListBox.css";
import MovieSnippet from "../MovieSnippet/MovieSnippet.jsx";

const MovieListBox = (props) => {
    const {
        movieListData = [],
        title,
        
    } = props;

    const getSnippets = () => {
        const snippets = [];
        if (movieListData) {
            for (let idx = 0; idx < movieListData.length && idx < 10; idx++){
                const item = movieListData[idx];
                snippets.push(
                    <MovieSnippet snippetData={item} key= {idx}></MovieSnippet>
                )
            }
            return snippets;
        }
    }
    return (
        <div className="movie-list-box-container">
            <p className="movie-list-box-title">
                {title}
            </p>
            <div className="snippet-list-wrapper">
                {getSnippets()}
            </div>
        </div>
    )

}

export default MovieListBox;