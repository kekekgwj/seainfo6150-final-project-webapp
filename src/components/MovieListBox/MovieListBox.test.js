import React from "react";
import MovieListBox from "./MovieListBox";

const movieList = [
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/2b72a4da-80ba-4b58-9459-4273f9089d2e.jpg",
      isNew: false,
      rating: 9.2,
      name : "The Godfather",
      id: "2b72a4da-80ba-4b58-9459-4273f9089d2e",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
  {
      cover: "http://localhost:5000/images/poster/p2636551330.jpeg",
      isNew: true,
      rating: 7,
      name : "nobody",
      id: "p2636551330",
  },
];
describe("Movie List box tests", () => {
  it("renders correctly", () => {
    const { container } = render(<MovieListBox movieListData={movieList} title="test title"/>);
    expect(container).toMatchSnapshot();
  });
});