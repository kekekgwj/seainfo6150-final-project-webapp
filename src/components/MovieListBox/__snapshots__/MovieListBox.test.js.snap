// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie List box tests renders correctly 1`] = `
<div>
  <div
    class="movie-list-box-container"
  >
    <p
      class="movie-list-box-title"
    >
      test title
    </p>
    <div
      class="snippet-list-wrapper"
    >
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="2b72a4da-80ba-4b58-9459-4273f9089d2e"
            src="http://localhost:5000/images/poster/2b72a4da-80ba-4b58-9459-4273f9089d2e.jpg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            The Godfather
          </span>
          <span
            class="rating"
          >
            9.2
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
      <div
        class="movie-snippet-container"
      >
        <div
          class="cover-wrapper"
        >
          <img
            alt="movie-cover"
            id="p2636551330"
            src="http://localhost:5000/images/poster/p2636551330.jpeg"
          />
        </div>
        <p
          class="movie-snippet-info"
        >
          <span>
            <img
              alt="new-movie"
              class="new"
              src="new-icon.png"
            />
          </span>
          <span>
            nobody
          </span>
          <span
            class="rating"
          >
            7
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
